core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/workspacemanager/HttpWorkspaceManagerDAO.scala

```diff
@@ -64,6 +65,28 @@ class HttpWorkspaceManagerDAO(apiClientProvider: WorkspaceManagerApiClientProvid
         .stage(WorkspaceStageModel.MC_WORKSPACE)
     )
 
+  override def cloneWorkspace(sourceWorkspaceId: UUID,
+                              workspaceId: UUID,
+                              displayName: String,
+                              spendProfile: ProfileModel,
+                              ctx: RawlsRequestContext,
+                              location: Option[String]
+  ): CloneWorkspaceResult =
+    getWorkspaceApi(ctx).cloneWorkspace(
+      new CloneWorkspaceRequest()
+        .destinationWorkspaceId(workspaceId)
+        .displayName(displayName)
+        .spendProfile(spendProfile.toString)
```

Comment:
@ehigham this could be the source of the issue-- try passing `spendProfile.getId`.

Reply:
Yes, that allowed me to get a workspace (which displayed an error!). You should add unit test coverage for this method in `HttpWorkspaceManagerDAO` to make sure you are sending the correct args to WSM.

![image](https://user-images.githubusercontent.com/484484/209186867-54519fd4-171e-4459-8610-9688cca31fc9.png)

Reply:
Error shown is because we get back no resources for Request URL: https://workspace.dsde-dev.broadinstitute.org/api/workspaces/v1/bc1289e5-cb15-47f6-b117-96b494cb2df7/resources?stewardship=CONTROLLED&limit=1000
. Waiting to see if it resolves after some time….

Reply:
Ah good catch. That was a silly mistake!

Reply:
I'm surprised I didn't get something back from wsm complaining about this

Reply:
You did… it just was very buried!

Reply:
earlier I meant!

Reply:
Yeah, all the casting of strings to UUIDs is very suspect in WSM. Either declare them as UUIDs from the get-go, or at least add better error handling.

Reply:
Added a unit test. Hopefully I got it right again and didn't just code the test wrong too haha